addons/toolbars/src/components/ToolbarMenuToggle.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback, FC } from 'react';
+import { useGlobals } from '@storybook/api';
+import { ToolbarMenuButton } from './ToolbarMenuButton';
+import { withKeyboardCycle, WithKeyboardCycleProps } from '../hoc/withKeyboardCycle';
+import { getSelectedIcon } from '../utils/get-selected-icon';
+import { ToolbarMenuProps } from '../types';
+
+type ToolbarMenuToggleProps = ToolbarMenuProps & WithKeyboardCycleProps;
+
+export const ToolbarMenuToggle: FC<ToolbarMenuToggleProps> = withKeyboardCycle(
+  ({ id, name, description, cycleValues = [], toolbar: { title: _title, items, showName } }) => {
+    const [globals, updateGlobals] = useGlobals();
+
+    const currentValue = globals[id];
+    const hasGlobalValue = !!currentValue;
+    const icon = getSelectedIcon({ currentValue, items });
+
+    let title = _title;
+
+    // Deprecation support for old "name of global arg used as title"
+    if (showName && !title) {
+      title = name;
+    }
+
+    const setNext = useCallback(() => {
+      const currentIndex = cycleValues.indexOf(currentValue);
+      const currentIsLast = currentIndex === cycleValues.length - 1;
+
+      const newCurrentIndex = currentIsLast ? 0 : currentIndex + 1;
+      const newCurrent = cycleValues[newCurrentIndex];
+
+      updateGlobals({ [id]: newCurrent });
+    }, [currentValue, updateGlobals]);
+
+    return (
+      <ToolbarMenuButton
+        active={hasGlobalValue}
+        description={description}
+        onClick={setNext}
+        icon={icon}
```

Comment:
This assumes a default value is set, which is not always the case. I would like to see if there's a way to use the default toolbar icon, when no default value is present, rather than not displaying the button at all. Does this seem possible?

Reply:
Offset is - no icon, no button (well button, but 0px wide)

I do think it is not an unfair agreement to make with the users, that if you do not provide something to show, then we show nothing.

I can put in a default icon for when there is nothing though

Reply:
Misunderstood in previous comment - yes will look at that